game/theater/frontline.py

```diff
@@ -158,8 +158,11 @@ def point_from_a(self, distance: float) -> Point:
                 )
             else:
                 remaining_dist -= segment.attack_distance
-        raise RuntimeError(
-            f"Could not find front line point {distance} from {self.blue_cp}"
+        # Point was not found in any of the segment point_a:s. Returning the point from the last evaluated segment
+        # point_b, with heading towards the point_a, distance of (attack_distance - remaining_dist) meters.
+        return segment.point_b.point_from_heading(
+            segment.attack_heading.degrees - 180,
+            segment.attack_distance - remaining_dist,
```

Comment:
This is unrelated to the PR description and so belongs in a separate PR.

Beyond that

> Point was not found in any of the segment point_a:s

Why not? Isn't that what the check above is suppose to guarantee?

Reply:
No, that condition can be encountered e.g when the opposing force frontline units have been completely wiped out. In that situation _position_distance() will return the total attack_distance for that specific route, which is beyond the last point_a of the route (being actually at the point_b of the last segment).